shanoir-uploader/src/main/java/org/shanoir/downloader/ShanoirDownloader.java

```diff
@@ -205,35 +205,33 @@ private void initProperties(final String fileName, final Properties properties)
 		}
 	}
 
-	public static void saveResponseToFile(File destDir, HttpResponse response) throws IOException {
+	public static void saveResponseToFile(File destDir, CloseableHttpResponse response) throws IOException {
 		Header header = response.getFirstHeader(HttpHeaders.CONTENT_DISPOSITION);
 		String fileName = header.getValue();
 		fileName = fileName.replace("attachment;filename=", "");
-
 		final File downloadedFile = new File(destDir + "/" + fileName);
-
 		FileUtils.copyInputStreamToFile(response.getEntity().getContent(), downloadedFile);
 	}
 
 	public static String downloadDataset(File destDir, Long datasetId, String format,
 			ShanoirUploaderServiceClient shng) throws Exception {
 		System.out.println("Downloading dataset " + datasetId + "...");
-		HttpResponse response = shng.downloadDatasetById(datasetId, format);
+		CloseableHttpResponse response = shng.downloadDatasetById(datasetId, format);
```

Comment:
surround executions with try {} catch {} the close the response in finally{} clause ? (I don't really know to which degree  a CloseableHttpResponse HAS to be closed everytime or not..)

Reply:
Hi Jean-Come,
the main architecture principle here is to re-use the httpClient instance and not close it with each request.
From my point of view one instance of http client is responsible to manage its connections and take care
of all requests. This one instance shall only be closed, when ShUp shuts down. The only exception is the
proxyTest during application startup, where I do not know the target server yet (depends on profile selection).
In this case, I create and close a separate http client to check for the proxy connection.

BUT your comment had me have another look again on ShanoirUploaderServiceClient, and I missed some try statements, to release the connection, after the download-methods and fixed them, what should be more clean now.

With kind regards,
Michael